packages/run-protocol/test/psm/test-psm.js

```diff
@@ -0,0 +1,190 @@
+/// no @ts-check because AVA context is hard to type
+
+import { test } from '@agoric/zoe/tools/prepare-test-env-ava.js';
+
+import '@agoric/zoe/exported.js';
+import '../../src/vaultFactory/types.js';
+
+import path from 'path';
+import { E } from '@endo/eventual-send';
+import bundleSource from '@endo/bundle-source';
+import { makeFakeVatAdmin } from '@agoric/zoe/tools/fakeVatAdmin.js';
+import { makeZoeKit } from '@agoric/zoe';
+import { AmountMath, makeIssuerKit } from '@agoric/ertp';
+import { resolve as importMetaResolve } from 'import-meta-resolve';
+import { makeLoopback } from '@endo/captp';
+import { makeTracer } from '../../src/makeTracer.js';
+
+const pathname = new URL(import.meta.url).pathname;
+const dirname = path.dirname(pathname);
+
+const psmRoot = `${dirname}/../../src/psm/psm.js`;
+const trace = makeTracer('TestPSM');
+
+const makeBundle = async sourceRoot => {
+  const url = await importMetaResolve(sourceRoot, import.meta.url);
+  const contractBundle = await bundleSource(new URL(url).pathname);
+  console.log(`makeBundle ${sourceRoot}`);
+  return contractBundle;
+};
+
+const BASIS_POINTS = 10000n;
+const WantStableFeeBP = 1n;
+const GiveStableFeeBP = 3n;
+const MINT_LIMIT = 20_000_000n * 1_000_000n;
+
+/**
+ * Compute the fee in runBrand based on `given`. Choose the fee ratio
+ * appropriate to teh `given` brand.
+ *
+ * @param {Amount<'nat'>} given
+ * @param {Brand} stableBrand
+ * @returns {Amount<'nat'>}
+ */
+const minusFee = (given, stableBrand) => {
+  const feeBP = given.brand === stableBrand ? GiveStableFeeBP : WantStableFeeBP;
+  return AmountMath.make(
+    stableBrand,
+    BigInt((given.value * (BASIS_POINTS - feeBP)) / BASIS_POINTS),
+  );
+};
+
+// let testJig;
+const setJig = _jig => {
+  // testJig = jig;
+};
+
+/**
+ * @param {ExecuteContract} t
+ */
+test.before(async t => {
+  // makeBundle is slow, so we bundle each contract once and reuse in all tests.
+  const [psmBundle] = await Promise.all([makeBundle(psmRoot)]);
```

Comment:
does Promise.all add anything here?
```suggestion
  const psmBundle = await makeBundle(psmRoot);
```

Reply:
It's to keep the pattern from other tests that use multiple bundles. But can be deleted.